Source/Core/VideoCommon/IndexGenerator.cpp

```diff
@@ -182,77 +133,122 @@ u16* IndexGenerator::AddFan(u16* Iptr, u32 numVerts, u32 index)
 {
-  for (u32 i = 1; i < numVerts; i += 2)
+  for (u32 i = 1; i < num_verts; i += 2)
   {
-    *Iptr++ = index + i - 1;
-    *Iptr++ = index + i;
+    *index_ptr++ = index + i - 1;
+    *index_ptr++ = index + i;
   }
-  return Iptr;
+  return index_ptr;
 }
 
-// shouldn't be used as strips as LineLists are much more common
+// Shouldn't be used as strips as LineLists are much more common
 // so converting them to lists
-u16* IndexGenerator::AddLineStrip(u16* Iptr, u32 numVerts, u32 index)
+u16* AddLineStrip(u16* index_ptr, u32 num_verts, u32 index)
 {
-  for (u32 i = 1; i < numVerts; ++i)
+  for (u32 i = 1; i < num_verts; ++i)
   {
-    *Iptr++ = index + i - 1;
-    *Iptr++ = index + i;
+    *index_ptr++ = index + i - 1;
+    *index_ptr++ = index + i;
   }
-  return Iptr;
+  return index_ptr;
 }
 
-// Points
-u16* IndexGenerator::AddPoints(u16* Iptr, u32 numVerts, u32 index)
+u16* AddPoints(u16* index_ptr, u32 num_verts, u32 index)
 {
-  for (u32 i = 0; i != numVerts; ++i)
+  for (u32 i = 0; i != num_verts; ++i)
+  {
+    *index_ptr++ = index + i;
+  }
+  return index_ptr;
+}
+}  // Anonymous namespace
+
+void IndexGenerator::Init()
+{
+  if (g_Config.backend_info.bSupportsPrimitiveRestart)
+  {
+    m_primitive_table[OpcodeDecoder::GX_DRAW_QUADS] = AddQuads<true>;
+    m_primitive_table[OpcodeDecoder::GX_DRAW_QUADS_2] = AddQuads_nonstandard<true>;
+    m_primitive_table[OpcodeDecoder::GX_DRAW_TRIANGLES] = AddList<true>;
+    m_primitive_table[OpcodeDecoder::GX_DRAW_TRIANGLE_STRIP] = AddStrip<true>;
+    m_primitive_table[OpcodeDecoder::GX_DRAW_TRIANGLE_FAN] = AddFan<true>;
+  }
+  else
   {
-    *Iptr++ = index + i;
+    m_primitive_table[OpcodeDecoder::GX_DRAW_QUADS] = AddQuads<false>;
+    m_primitive_table[OpcodeDecoder::GX_DRAW_QUADS_2] = AddQuads_nonstandard<false>;
+    m_primitive_table[OpcodeDecoder::GX_DRAW_TRIANGLES] = AddList<false>;
+    m_primitive_table[OpcodeDecoder::GX_DRAW_TRIANGLE_STRIP] = AddStrip<false>;
+    m_primitive_table[OpcodeDecoder::GX_DRAW_TRIANGLE_FAN] = AddFan<false>;
   }
-  return Iptr;
+  m_primitive_table[OpcodeDecoder::GX_DRAW_LINES] = AddLineList;
+  m_primitive_table[OpcodeDecoder::GX_DRAW_LINE_STRIP] = AddLineStrip;
+  m_primitive_table[OpcodeDecoder::GX_DRAW_POINTS] = AddPoints;
 }
 
-u32 IndexGenerator::GetRemainingIndices()
+void IndexGenerator::Start(u16* index_ptr)
 {
-  u32 max_index = 65534;  // -1 is reserved for primitive restart (ogl + dx11)
-  return max_index - base_index;
+  m_index_buffer_current = index_ptr;
+  m_base_index_ptr = index_ptr;
+  m_base_index = 0;
+}
+
+void IndexGenerator::AddIndices(int primitive, u32 num_vertices)
+{
+  m_index_buffer_current =
+      m_primitive_table[primitive](m_index_buffer_current, num_vertices, m_base_index);
+  m_base_index += num_vertices;
+}
+
+void IndexGenerator::AddExternalIndices(const u16* indices, u32 num_indices, u32 num_vertices)
+{
+  std::memcpy(m_index_buffer_current, indices, sizeof(u16) * num_indices);
```

Comment:
```suggestion
  std::memcpy(m_index_buffer_current, indices, sizeof(indices[0]) * num_indices);
```
If you ever need u32 indices it'll make migration a bit easier.

Reply:
This is unrelated to the actual changes being made.

Reply:
I respectfully disagree - you've changed destination from `index_buffer_current` to `m_index_buffer_current` so you "touched" this line in a meaningful way in this PR already.

Reply:
Yes. I've changed the line because we follow a naming convention for member variables in classes. This does not mean I also need to resolve everything else related to that line within an unrelated PR. The purpose of this change is to eliminate static state in the class, which it does. It does not try to address anything else with the interface.

Reply:
Yes. I've changed the line because we follow a naming convention for member variables in classes. This does not mean I also need to resolve everything else related to that line within an unrelated PR. The purpose of this change is to eliminate static state in the class, which it does. It does not try to address anything else with the interface.